src/myrtlespeech/data/sampler.py

```diff
@@ -1,14 +1,37 @@
 import random
 
 
-class RandomBatchSampler:
-    """TODO"""
+class SequentialRandomSampler:
+    """A sequential or random iterable over batches.
+    The iterator used each time this iterable is iterated over will yield
+    batches either sequentially (i.e. in-order) or randomly (uniform without
+    replacement) from `batches`.
+    This iterable records the number of times it has returned an iterator. A
+    sequential iterator is returned if the current count is in `sequential`.
+    Args:
+        batches: A list of batches.
+        n_iterators (optional, int): Number of iterators returned so far.
+        sequential (optional, set of int): Counts at which to return a
+            sequential iterator.
```

Comment:
These are the args for `__iter__`? Add init args (`indices, batch_size, shuffle, drop_last, n_iterators, sequential`)?